src/org/zaproxy/zap/extension/ascanrulesAlpha/ElmahScanner.java

```diff
@@ -0,0 +1,182 @@
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+import org.parosproxy.paros.Constant;
+
+import java.io.IOException;
+
+/**
+ * Active scan rule which checks whether or not elmah.axd is exposed.
+ * https://github.com/zaproxy/zaproxy/issues/3279
+ * 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class ElmahScanner extends AbstractHostPlugin {
+
+	private static final String MESSAGE_PREFIX = "ascanalpha.elmahscanner.";
+	private static final int PLUGIN_ID = 40028;
+
+	private static final Logger LOG = Logger.getLogger(ElmahScanner.class);
+
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	private String getOtherInfo() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "otherinfo");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		return technologies.includes(Tech.IIS) || technologies.includes(Tech.Windows) || technologies.includes(Tech.ASP)
+				|| technologies.includes(Tech.MsSQL);
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INFO_GATHER;
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		return 215; // CWE-215: Information Exposure Through Debug Information
+	}
+
+	@Override
+	public int getWascId() {
+		return 13; // WASC-13: Informatin Leakage
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+
+		// Check if the user stopped things. One request per URL so check before
+		// sending the request
+		if (isStop()) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Scanner " + getName() + " Stopping.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		newRequest.getRequestHeader().setMethod(HttpRequestHeader.GET);
+		URI baseUri = getBaseMsg().getRequestHeader().getURI();
+		URI elmahUri = null;
+		try {
+			elmahUri = new URI(baseUri.getScheme(), null, baseUri.getHost(), baseUri.getPort(), "/elmah.axd");
+		} catch (URIException uEx) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("An error occurred creating a URI for the: " + getName() + " scanner. " + uEx.getMessage(),
+						uEx);
+				return;

```

Comment:
Ah, just noticed that the return is inside the if not outside (same for following return).